### ISSUE TITLE
accelerator.sync_gradients

### ISSUE THREAD

**Issue by AriesJin** [huggingface/trl#3988]
### Reproduction

```python
from trl import ...
python -i examples/scripts/ppo/ppo.py \
    --dataset_name trl-internal-testing/descriptiveness-sentiment-trl-style \
    --dataset_train_split descriptiveness \
    --learning_rate 3e-6 \
    --output_dir Qwen2.5-0.5B-Instruct_ppo \
    --per_device_train_batch_size 2 \
    --gradient_accumulation_steps 2 \
    --total_episodes 10000 \
    --model_name_or_path EleutherAI/pythia-160m \
    --missing_eos_penalty 1.0
```

### System Info

Why in the PPO_trainer, instead of executing accelerator.sync_gradients before updating gradients, does it update gradients every time?
[code](https://github.com/huggingface/trl/blob/main/trl/trainer/ppo_trainer.py#L586)
```
logprobs_diff = new_logprobs - mb_logprobs
ratio = torch.exp(logprobs_diff)
pg_losses = -mb_advantage * ratio
pg_losses2 = -mb_advantage * torch.clamp(ratio, 1.0 - args.cliprange, 1.0 + args.cliprange)
pg_loss_max = torch.max(pg_losses, pg_losses2)
pg_loss = masked_mean(pg_loss_max, ~padding_mask[micro_batch_inds])
loss = pg_loss + args.vf_coef * vf_loss
accelerator.backward(loss)
optimizer.step()
optimizer.zero_grad()
```

### Checklist

- [x] I have checked that my issue isn't already filed (see [open issues](https://github.com/huggingface/trl/issues?q=is%3Aissue))
- [x] I have included my system information
- [x] Any code provided is minimal, complete, and reproducible ([more on MREs](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any code provided is properly formatted in code blocks, (no screenshot, [more on code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any traceback provided is complete

**Comment by faker52** [comment]
I meet the same question

**Comment by LeonEricsson** [comment]
Gradient synchronization is handled by the gradient accumulation wrapper; avoiding communication during accumulation.

https://github.com/huggingface/trl/blob/edbe8234bc7e528f72ac76607de9d3e4753e2709/trl/trainer/ppo_trainer.py#L548

See [docs](https://huggingface.co/docs/accelerate/en/usage_guides/gradient_accumulation) for further details

**Comment by AriesJin** [comment]
> Gradient synchronization is handled by the gradient accumulation wrapper; avoiding communication during accumulation.
> 
> [trl/trl/trainer/ppo_trainer.py](https://github.com/huggingface/trl/blob/edbe8234bc7e528f72ac76607de9d3e4753e2709/trl/trainer/ppo_trainer.py#L548)
> 
> Line 548 in [edbe823](/huggingface/trl/commit/edbe8234bc7e528f72ac76607de9d3e4753e2709)
> 
>  with accelerator.accumulate(model): 
> See [docs](https://huggingface.co/docs/accelerate/en/usage_guides/gradient_accumulation) for further details

got it, thanks
